client/src/app/unauthorized.component.test.tsx

```diff
@@ -0,0 +1,25 @@
+import { describe, expect, it } from 'vitest';
+
+import { render, screen, userEvent } from '~/test/utils';
+
+import { Unauthorized } from './unauthorized.component';
+
+describe('Unauthorized', () => {
+  it('should render unauthorized component', () => {
+    render(<Unauthorized />);
+
+    expect(screen.getByRole('heading', { name: /unauthorized/i })).toBeInTheDocument();
+    expect(screen.getByText(/you do not have access to the requested page./i)).toBeInTheDocument();
+    expect(screen.getByRole('button', { name: /go back/i })).toBeInTheDocument();
+  });
+
+  it('should move one step backward from unauthorized page when clicks on the "Go Back" button', async () => {
+    const user = userEvent.setup();
+    const goBack = vi.fn();
+    render(<Unauthorized goBack={goBack} />);
+
+    await user.click(screen.getByRole('button', { name: /go back/i }));
```

Comment:
For things like search terms or forms, where we'd be lowercasing everything anyway I imagine it would be redundant to make matchers case-insensitive.

But for things like buttons, typos could slip through. Like if I typed `go Back` as the button text, this test wouldn't catch the mistake because of the case-insensitive regex.

Reply:
What matters most is we are consistent. I don't want some using plain strings and others using regex, do what the docs suggets, don't make it up as you go along @RoryMacGregor88. If you disagree, fine, state reasons, argue over it and agree, but don't arbitrarily, do things different just cause you disagree.